src/frontend/src/session/cursor_manager.rs

```diff
@@ -111,7 +112,12 @@ impl Cursor {
         formats: &Vec<Format>,
     ) -> Result<(Vec<Row>, Vec<PgFieldDescriptor>)> {
         match self {
-            Cursor::Subscription(cursor) => cursor.next(count, handle_args, formats).await,
+            Cursor::Subscription(cursor) => {
+                cursor.next(count, handle_args, formats).await.map_err(|e| {
+                    cursor.cursor_metrics.subscription_cursor_error_count.inc();
```

Comment:
How about updating the metric in `fetch_cursor.rs` just like what we did in `declare_cursor.rs`?

Reply:
Since we don't distinguish between `cursor `and `subscription cursor` until `next()`, we can only do it here. So we cannot correctly add to the `subscription_cursor_fetch_err `counts in` fetch_cursor.rs`